### PR TITLE
Tests config for serialization of arrays

### PR DESCRIPTION
**Pull request by Karm**
* Depends on https://github.com/oracle/graal/issues/8509 being released via https://github.com/graalvm/graalvm-community-jdk21u/pull/10

* Depends on https://github.com/oracle/graal/issues/9581 being fixed, backported and released.

Merging it now would trigger failures for CI.

### CODE REVIEW

**Comment by Karm** [comment]
```
 > but was: <Exception in thread "main" com.oracle.svm.core.jdk.UnsupportedFeatureError: SerializationConstructorAccessor class not found for declaringClass: [Z (targetConstructorClass: java.lang.Object). Usually adding [Z to serialization-config.json fixes the problem.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:121)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.serialize.SerializationSupport.getSerializationConstructorAccessor(SerializationSupport.java:164)
	at java.base@22.0.2/jdk.internal.reflect.SerializationConstructorAccessorGenerator.generateSerializationConstructor(SerializationConstructorAccessorGenerator.java:51)
	at java.base@22.0.2/jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:384)
	at java.base@22.0.2/jdk.internal.reflect.ReflectionFactory.newConstructorForSerialization(ReflectionFactory.java:288)
	at jdk.unsupported@22.0.2/sun.reflect.ReflectionFactory.newConstructorForSerialization(ReflectionFactory.java:100)
	at for_serialization.Main.main(Main.java:147)
	at java.base@22.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
>
[INFO] 
Error:  Tests run: 9, Failures: 1, Errors: 0, Skipped: 0
```

Expected. The fix is not in the released JDK 22 Mandrel

File: testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java [hunk 1]
```diff
@@ -818,6 +817,68 @@ public void monitorFieldOffset(TestInfo testInfo, Apps app) throws IOException,
...
 
+    @Test
+    @Tag("builder-image")
+    @IfMandrelVersion(min = "23.1.5", inContainer = true)
+    public void forSerializationContainerTest(TestInfo testInfo) throws IOException, InterruptedException {
+        forSerialization(testInfo, Apps.FOR_SERIALIZATION_BUILDER_IMAGE);
+    }
+
+    @Test
+    @IfMandrelVersion(minJDK = "23.1.5")
```
Comment (zakkak) [inline testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java:828]: This should be:

```suggestion
    @IfMandrelVersion(min = "23.1.5")
```

no?

**Review by zakkak — CHANGES_REQUESTED** [review]
> Expected. The fix is not in the released JDK 22 Mandrel

And it's not going to get backported to it, so we probably need to disable it in that configuration.

**Comment by Karm** [comment]
> > Expected. The fix is not in the released JDK 22 Mandrel
> 
> And it's not going to get backported to it, so we probably need to disable it in that configuration.

I see, indeed. Lemme update it...

**Comment by zakkak** [comment]
> or it's parsing Mandrel 24 as JDK 24?

The logs say
```
The test suite runs with Mandrel version 24.0.2 in container, JDK 22.0.2.
```

so it doesn't look like a parsing issue. 

I think the issue is that you are passing "21.999" as the `maxJDK` which doesn't match the pattern in https://github.com/Karm/mandrel-integration-tests/blob/9f6c895eb2de6a0b8dae8e50d826f31ee2b70606/testsuite/src/it/java/org/graalvm/tests/integration/utils/versions/MandrelVersionCondition.java#L46

**Comment by zakkak** [comment]
@Karm thanks for the fix. I opened https://github.com/Karm/mandrel-integration-tests/issues/280 to keep track of the version parsing issue.

**Review by zakkak — APPROVED** [review]

**Pull request merged**